draft-ietf-webtrans-http3.md

```diff
@@ -311,10 +311,46 @@ an implementation to choose what stream or datagram to discard.
 
 # Session Termination
 
-An WebTransport session over HTTP/3 is terminated when either endpoint closes
-the stream associated with the CONNECT request that initiated the session.
-Upon learning about the session being terminated, the endpoint MUST stop
-sending new datagrams and reset all of the streams associated with the session.
+An WebTransport session over HTTP/3 is considered terminated when either
+endpoint closes the stream associated with the CONNECT request that initiated
+the session.  Upon learning about the session being terminated, the endpoint
+MUST stop sending new datagrams and reset all of the streams associated with
+the session.
+
+To terminate a session with a detailed error message, an application MAY send
+an HTTP capsule [HTTP3-DATAGRAM] of type CLOSE_WEBTRANSPORT_SESSION (0x2843).
+The format of the capsule SHALL be as follows:
+
+~~~
+CLOSE_WEBTRANSPORT_SESSION Capsule {
+  Application Error Code (64),
+  Application Error Message Length (i),
+  Application Error Message (..1024),
+}
+~~~
+
+CLOSE_WEBTRANSPORT_SESSION has the following fields:
+
+Application Error Code:
+
+: A 64-bit error code provided by the application closing the connection.
+
+Application Error Message:
+
+: An error message string provided by the application closing the connection.
```

Comment:
What's the purpose of this message? Is it exposed to JavaScript?

Reply:
Should we explicitly state it's a UTF-8 encoded string? I see that in CONNECTION_CLOSE. https://datatracker.ietf.org/doc/html/rfc9000#section-19.19

Reply:
Clarified that this is UTF-8.